dagmc_stats.py

```diff
@@ -48,6 +44,186 @@ def get_entity_ranges(my_core, meshset, entity_types):
     for entity_type in entity_types:
         entity_ranges[entity_type] = my_core.get_entities_by_type(meshset, entity_type) 
-
+        if entity_type == 11:
+            entity_ranges['Volumes'] = my_core.get_entities_by_type_and_tag(meshset, types.MBENTITYSET, dagmc_tags['geom_dim'], [3])
+            entity_ranges['Surfaces'] = my_core.get_entities_by_type_and_tag(meshset, types.MBENTITYSET, dagmc_tags['geom_dim'], [2])
+            entity_ranges['Curves'] = my_core.get_entities_by_type_and_tag(meshset, types.MBENTITYSET, dagmc_tags['geom_dim'], [1])
     return entity_ranges
 
+def get_surfaces_per_volume(my_core, entity_ranges):
+    """
+    Get the number of surfaces that each volume in a given file contains
+    
+    inputs
+    ------
+    my_core : a MOAB core instance
+    entity_ranges : a dictionary of the ranges of each tag in a file
+    
+    outputs
+    -------
+    pdf : a histogram that displays the number of surfaces each volume has (matplotlib)
+    """
+    freqs = []
+    for volumeset in entity_ranges['Volumes']:
+        freqs.append(my_core.get_child_meshsets(volumeset).size())
+    stats = {}
+    stats['minimum'] = min(freqs)
+    stats['maximum'] = max(freqs)
+    stats['median'] = np.median(freqs)
+    stats['average'] = np.average(freqs)
+    return stats
+
+def make_verts_dictionary(vert_list):
+    """
+    returns a dictionary of frequency of the values of the triangles per vertex (ie {number of triangles per vertex: occurences})
+    inputs
+    -----
+    vert_list : a list of all vertices in a file
+    
+    outputs:
+    --------
+    sorted_dict: a dictionary that has information about triangles per vertex, sorted by its keys.
+    """
+    vertices_and_frequencies = {} 
+    #makes dictionary with each vertex and the number of triangles on it
+    for vert in vert_list:
+        number_of_vertex = vert
+        try:
+            vertices_and_frequencies[number_of_vertex] += 1
+        except KeyError:
+            vertices_and_frequencies[number_of_vertex] = 1
+    number_of_frequencies = {}
+    #makes dictionary with the number of triangles, and the number of vertices that have that number
+    for vertex in vertices_and_frequencies:
+        try:
+            number_of_frequencies[vertices_and_frequencies[vertex]] += 1
+        except KeyError:
+            number_of_frequencies[vertices_and_frequencies[vertex]] = 1
+    sorted_dict = {}
+    #sorts the dictionary
+    for frequency in sorted(number_of_frequencies.keys()):
+        sorted_dict[frequency] = number_of_frequencies[frequency]
+    return(sorted_dict)
+def find_median(sorted_dict):
+    """
+    finds the median of a dictionary with format {value:occurences}
+    
+    inputs
+    ------
+    sorted_dict : a dictionary sorted by its keys
+    
+    outputs
+    -------
+    mean : the median of the dataset
+    
+    example dictionary:
+    {3:2, 4:5, 6:3} = [3,3,4,4,4,4,4,6,6,6]
+    """
+    #finds the number of data points
```

Comment:
Suggest to use `len(sorted_dict)` to get the number of data points.

Reply:
If you look at the example dictionary, the length of the dictionary isn't what I'm after. Instead, I want to find the length of the subsequent list. For the example, `len(example_dict) = 3`, but the value I want is 2+5+3 = 10 (or `example_dict[3] + example_dict[4] +example_dict[5]`).

Reply:
Got it. Then I suggest to try:
```python
length = sum(sorted_dict.values())
```
Eg:
```python
>>> d = {3:2, 4:5, 6:3}
>>> sum(d.values())
10
```